### PR TITLE
Adds interface file for converting from pycapnp schema to compiled in sc...

### LINKED ISSUE #1794: Enable passing pycapnp builders/readers to C++ code
To integrate the Python and C++ serialization code, we need to be able to pass a builder or reader for some part of a Cap'n Proto struct from the Python code to the C++ code.

Part of #1449

### PR DESCRIPTION
**Pull request by scottpurdy**
...hema and uses it with SWIGed C++ SpatialPooler class's read and write methods.

fixes #1794 

@oxtopus please review

### CODE REVIEW

**Comment by rhyolight** [comment]
Linux build failures look legit.

**Comment by oxtopus** [comment]
https://github.com/scottpurdy/nupic/pull/5 seems to work for me.

**Comment by scottpurdy** [comment]
@oxtopus - can I merge?

**Comment by oxtopus** [comment]
:+1:

File: setup.py [hunk 1]
```diff
@@ -341,11 +341,19 @@ def getExtensionModules(nupicCoreReleaseDir, platform, bitness):
   commonLinkFlags = [
     "-m" + bitness,
     "-fPIC",
-    "-L" + pythonPrefix + "/lib"]
+    "-L" + pythonPrefix + "/lib",
+    "-L" + nupicCoreReleaseDir + "/lib",
```
Comment (rhyolight) [inline setup.py]: Just curious, why did you add this line? Does it have anything to do with the rest of the PR?

Comment (scottpurdy) [inline setup.py]: The three lines below specify libraries to link to and this line makes sure it looks in the right place.

Comment (rhyolight) [inline setup.py]: Make sense, thanks.
